docs[MG]: Traceability Matrix

jawadsalwati opened the pull request:
<!--
Name your PR using the conventional commits specification: https://www.conventionalcommits.org/en/v1.0.0/#specification.
  - feat: new feature
  - fix: bug fix
  - docs: documentation only changes
  - style: formatting, no semantic meaning change to code, e.g., semicolons, white-space
  - refactor: refactoring production code, neither fixes bug nor adds feature, e.g., renaming a signature
  - perf: code change that improves performance
  - test: adding missing tests or correcting existing tests
  - chore: "busy-work" tasks, don't affect code or documentation per se
  - build: changes that affect the build system or external dependencies, e.g., pipenv, poetry
  - ci: changes to configuration files and scripts, e.g., travis, GH actions
  - revert: revert a previous commit (header should be header of reverted commit)
-->

## Description
<!--Describe the purpose of this pull request.-->
Completes #287 .

## Changes
<!--List the changes you have made.-->
Completed MG-8.1 Traceability Matrix. 

## Additional Information
<!--Include any additional information, such as how to test your changes.-->

## Assignee Checklist
<!--Checklist of things that assignee completed (or has to complete if PR is still a draft)-->
- [x] The commits are named in accordance with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#specification).
- [x] The PR is named in accordance with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#specification).
- [x] There are no future/additional changes required. If there are, they are mentioned in *Additional Information* above.
- [x] I have changed the issue status in the [project tracker](https://github.com/users/AhmedAl-Hayali/projects/1/) to *In Review*.
- [x] I have created a review issue in the [project tracker](https://github.com/users/AhmedAl-Hayali/projects/1/) and assigned it to my reviewer.
- [x] I have provided my reviewer a checklist of necessary actions, if applicable.

## Reviewer Checklist
<!--Checklist of things that reviewer should do/have a look at (optional, feel free to an "N/A" instead of checklist items)-->
- [ ] I have updated the issue(s) #370   status(es) on the [project tracker](https://github.com/users/AhmedAl-Hayali/projects/1/).

chenans1 reviewed (changes requested):
Generally, it feels like there are extra modules listed for some of the requirements. 

Here are a few examples I and Matthew felt needed adjustments: 

For R1, it seems more appropriate that the search query module instead of the client communication module should be linked there, as the user needs to directly interact with the search function and not necessarily the client communication. 

R2 - the client communication module should not be there. It is probably more appropriate that only the GUI and results module should be linked. As R2 is worded in the SRS, it only describes displaying results and not sending data between modules. 

R6 - this should only be the search query module as that is only module constructing the search request, sending the information to the backend is related to another requirement. 

R11 - should only be server and client, as the requirement in the SRS only states "Receive".

jawadsalwati commented:
@chenans1
Summary of Changes Based on Feedback:
R1: Removed Client-Side Communication Module and replaced it with Search Query Module, since the focus is on search functionality.
R2: Removed Client-Side Communication Module as displaying search results does not involve communication.
R4: Removed Client-Side Communication Module, as file validation happens within the Audio File Module.
R6: Removed Spotify Module and Client-Side Communication Module, as constructing queries is solely the responsibility of the Search Query Module.
R11: Removed Results Display Module and GUI Module, as the requirement focuses only on the receipt of results.
